### PR TITLE
add tomcat user:password to curl command

### PR DESCRIPTION
**Pull request by jgeromero**
for issue #381

### CODE REVIEW

**Comment by ktsaou** [comment]
thanks!

2 questions:

1. Did it work if username and password were given at the url? If it did, then we shouldn't change it.
2. The way it is now, if username and password are empty, will it work? It will run `curl -u : http://...`

**Comment by jgeromero** [comment]
Hi @ktsaou,

1. Yes it did, but credentials need to be passed in the url as http://user:pass@localhost..
2. It will not work if they're left blank, since tomcat requires a valid username to view status page. 

In both cases, valid credentials need to be specified somewhere.. I kind of like having variables curl can use as opposed to passing those credentials in line with the request. 

Thank you,

**Comment by ktsaou** [comment]
ok.

Could you please check if any of the 3 variables is empty and in case any of them is empty, show an error message to instruct the user to set all 3 variables?

**Comment by jgeromero** [comment]
Ok changes made and committed. Thanks!

File: charts.d/tomcat.chart.sh [hunk 1]
```diff
@@ -25,10 +30,27 @@ tomcat_check() {
 
 	require_cmd xmlstarlet || return 1
 
+
+	# check if url, username, passwords are set
+	if [ -z "${tomcat_url}" ]; then
+	  echo "tomcat url is unset or set to the empty string"
```
Comment (ktsaou) [inline charts.d/tomcat.chart.sh]: Please send this to `>&2`

File: charts.d/tomcat.chart.sh [hunk 2]
```diff
@@ -25,10 +30,27 @@ tomcat_check() {
...
+		return 1
+	fi
+	if [ -z "${tomcatUser}" ]; then
+    	  echo "tomcat user is unset or set to the empty string"
```
Comment (ktsaou) [inline charts.d/tomcat.chart.sh]: Please send this to `>&2`

File: charts.d/tomcat.chart.sh [hunk 3]
```diff
@@ -25,10 +30,27 @@ tomcat_check() {
...
+		return 1
+	fi
+	if [ -z "${tomcatPassword}" ]; then
+    	  echo "tomcat password is unset or set to the empty string"
```
Comment (ktsaou) [inline charts.d/tomcat.chart.sh]: Please send this to `>&2`
Otherwise it will got to netdata

**Comment by ktsaou** [comment]
one last thing. Use `echo >&2` or the output will go to netdata.
Thanks.

**Comment by jgeromero** [comment]
All set. Thanks.

**Pull request merged**